src/main/scala/nl/wbaa/gargoyle/proxy/route/GetRoute.scala

```diff
@@ -1,28 +1,20 @@
 package nl.wbaa.gargoyle.proxy.route
 
+import akka.http.scaladsl.model.StatusCodes
 import akka.http.scaladsl.server.Directives._
 import com.typesafe.scalalogging.LazyLogging
-import nl.wbaa.gargoyle.proxy.providers.StorageProvider
+import nl.wbaa.gargoyle.proxy.handler.RequestHandler
 import nl.wbaa.gargoyle.proxy.route.CustomDirectives._
-import nl.wbaa.gargoyle.proxy.response.ResponseHelpers._
 
-import scala.util.{ Failure, Success, Try }
-
-case class GetRoute()(implicit provider: StorageProvider) extends LazyLogging {
+case class GetRoute() extends LazyLogging with RequestHandler {
 
   // accepts all GET requests
   def route() =
-    validateToken { tokenOk =>
+    checkPermission { secretKey =>
       get {
         extractRequestContext { ctx =>
-          Try(provider.translateRequest(ctx.request)) match {
-            case Success(s3Response) =>
-              stringComplete(s3Response)
-            case Failure(ex) => // all exceptions for now
-              throw new Exception(ex.getMessage)
-          }
+          redirect(ctx.request.uri.withAuthority("192.168.56.10", 8080), StatusCodes.TemporaryRedirect)
```

Comment:
Can we make the address where ceph runs configurable in the application.conf?

Reply:
@nielsdenissen looks like signature error appears when request is formed by aws s3api (I was able to reproduce it as well). Before, I did test that from sdk and it worked. From debug I see that request by sdk looks different that one from aws cmd.  

Also please add your code as well, let's test it and see...